include/meltpooldg/evaporation/evaporation_operation.hpp

```diff
@@ -0,0 +1,210 @@
+    void
+    reinit()
+    {
+      scratch_data->initialize_dof_vector(evaporation_velocity, vel_hanging_nodes_dof_idx);
+    }
+
+    void
+    solve()
+    {
+      level_set.update_ghost_values();
+      normal_vector.update_ghost_values();
+      reinit();
+      evaporation_velocities.resize(scratch_data->get_matrix_free().n_cell_batches() * dim *
+                                    scratch_data->get_matrix_free().get_n_q_points(ls_quad_idx));
+
+      FECellIntegrator<dim, 1, double> ls(scratch_data->get_matrix_free(), ls_dof_idx, ls_quad_idx);
+
+      FECellIntegrator<dim, dim, double> normal_vec(scratch_data->get_matrix_free(),
+                                                    normal_dof_idx,
+                                                    ls_quad_idx);
+
+      for (unsigned int cell = 0; cell < scratch_data->get_matrix_free().n_cell_batches(); ++cell)
+        {
+          Tensor<1, dim, VectorizedArray<double>> *evapor_velocity = begin_interface_velocity(cell);
+
+          ls.reinit(cell);
+          ls.read_dof_values_plain(level_set);
+          ls.evaluate(true, false);
+
+          normal_vec.reinit(cell);
+          normal_vec.read_dof_values_plain(normal_vector);
+          normal_vec.evaluate(true, false);
+
+          for (unsigned int q_index = 0; q_index < ls.n_q_points; ++q_index)
+            {
+              //@todo: comment
+              auto is_one_phase =
+                (evaporation_data.ls_value_gas == 1.0) ?
+                  UtilityFunctions::heaviside(ls.get_value(q_index), 0.0) :
+                  std::abs(1. - UtilityFunctions::heaviside(ls.get_value(q_index), 0.0));
+
+              auto density =
+                evaporation_data.density_liquid +
+                (evaporation_data.density_gas - evaporation_data.density_liquid) * is_one_phase;
+
+              evapor_velocity[q_index] =
+                normal_vec.get_value(q_index) * evaporation_data.evaporative_mass_flux / density;
+
+              if (evaporation_data.ls_value_gas == 1.0)
+                evapor_velocity[q_index] *= -1.0;
+            }
+        }
+
+      level_set.zero_out_ghosts();
+      normal_vector.zero_out_ghosts();
+
+      UtilityFunctions::fill_dof_vector_from_cell_operation_vec<dim, dim>(
+        evaporation_velocity,
+        scratch_data->get_matrix_free(),
+        vel_hanging_nodes_dof_idx,
+        ls_quad_idx,
+        scratch_data->get_fe(vel_hanging_nodes_dof_idx).tensor_degree(),     // fe_degree
+        scratch_data->get_fe(vel_hanging_nodes_dof_idx).tensor_degree() + 1, // n_q_points_1d
+        [&](const unsigned int cell,
+            const unsigned int quad) -> const Tensor<1, dim, VectorizedArray<double>> & {
+          return begin_interface_velocity(cell)[quad];
+        });
+
+      scratch_data->get_constraint(vel_hanging_nodes_dof_idx).distribute(evaporation_velocity);
+    }
+
+    inline Tensor<1, dim, VectorizedArray<double>> *
+    begin_interface_velocity(const unsigned int macro_cell)
```

Comment:
```suggestion
    begin_evaporation_velocity(const unsigned int macro_cell)
```

Maybe also define that is defined in the whole domain and not just on the "interface".